### DISCUSSION TITLE
Is @graphql-codegen/typescript-urql-graphcache disappeared?

### DISCUSSION THREAD

**Discussion by frederikhors** [urql-graphql/urql#2323] [Q&A] [answered]
I'm using `@graphql-codegen/typescript-urql-graphcache@2.2.6`. But I cannot find it anymore on https://www.graphql-code-generator.com/plugins/typescript-urql-graphcache.

Am I wrong?

What should I use?

**Comment by frederikhors** [comment]
I see it here: https://github.com/dotansimha/graphql-code-generator/tree/master/packages/plugins/typescript/urql-graphcache.

**Reply by Psvensso** [reply]
dead link

**Reply by mwahlhuetter** [reply]
seems like it has been moved to the community repo:
[https://github.com/dotansimha/graphql-code-generator-community/tree/main/packages/plugins/typescript/urql-graphcache](https://github.com/dotansimha/graphql-code-generator-community/tree/main/packages/plugins/typescript/urql-graphcache)

**Comment by JoviDeCroock** [comment, marked as answer]
Better to ask them as that is not our repo 😅

**Comment by productdevbook** [comment]
it's not in the documentation, I can't make sense of it. Is this still valid

**Reply by kitten** [reply]
The package is still around, albeit unmaintained: https://npm.im/@graphql-codegen/typescript-urql-graphcache
The source can now be found here: https://github.com/dotansimha/graphql-code-generator-community/tree/main/packages/plugins/typescript/urql-graphcache

I'm actually not sure if we ever had documentation for this.
It was experimental and unfinished and since we'd have to in the future maintain something like this with custom tooling, imho, I don't really want to encourage usage.

That said, the plugin generates a type that can be passed to Graphcache's `cacheExchange` as a generic type.

**Reply by frederikhors** [reply]
@kitten this is needed. Please, don't abandon us typescript generators! LOL 🤣

**Reply by kitten** [reply]
> we'd have to in the future maintain something like this with custom tooling

I mean, I'm not implying this won't happen, but I currently don't have a timeline for this, as it's not trivial to come up with good tooling for TypeScript. They basically don't make it easy to integrate something into `tsserver`

**Reply by productdevbook** [reply]
Thank you this is really necessary for us. It's working great now, please don't give up on it.

**Reply by frederikhors** [reply]
@kitten is there any alternative to `typescript-urql-graphcache`?

**Reply by nephix** [reply]
I'm also looking for an alternative for `typescript-urql-graphcache` but there doesn't seem to be any
